tests/test_websocket_writer.py

```diff
@@ -106,3 +108,60 @@ async def test_send_compress_text_per_message(protocol: Any, transport: Any) ->
     writer.transport.write.assert_called_with(b"\x81\x04text")
     await writer.send(b"text", compress=15)
     writer.transport.write.assert_called_with(b"\xc1\x06*I\xad(\x01\x00")
+
+
+@mock.patch("aiohttp.http_websocket.WEBSOCKET_MAX_SYNC_CHUNK_SIZE", 16)
+async def test_concurrent_messages_with_executor(protocol: Any, transport: Any) -> None:
+    """Ensure messages are compressed correctly when there are multiple concurrent writers.
+
+    This test generates messages large enough that they will
+    be compressed in the executor.
+    """
+    writer = WebSocketWriter(protocol, transport, compress=15)
+    queue: DataQueue[WSMessage] = DataQueue(asyncio.get_running_loop())
+    reader = WebSocketReader(queue, 50000)
+    writers = []
+    payloads = []
+    msg_length = 16 + 1
+    for count in range(1, 64 + 1):
+        payload = bytes((count,)) * msg_length
+        payloads.append(payload)
+        writers.append(writer.send(payload, binary=True))
+    await asyncio.gather(*writers)
+    for call in writer.transport.write.call_args_list:
+        call_bytes = call[0][0]
+        result, _ = reader.feed_data(call_bytes)
+        assert result is False
+        msg = await queue.read()
+        bytes_data: bytes = msg.data
+        assert len(bytes_data) == msg_length
+        assert bytes_data == bytes_data[0:1] * msg_length
+
+
+async def test_concurrent_messages_without_executor(
```

Comment:
I had this as a single test with params but I thought that would make it harder to understand what has gone wrong in the future so I split it into two tests since there are two different branches to test.  If there is a preference to have it a single test, I can put it back.

Reply:
If the only difference is the chunk size, then maybe just parametrize() it?

Reply:
Alternatively, I reckon all 3 could be combined by just parametrizing msg_length (with a callable, so it works with the 3rd case).

Reply:
Yup, that works